ledger-tool/src/main.rs

```diff
@@ -2350,6 +2478,72 @@ fn main() {
                         system_monitor_service.join().unwrap();
                     }
                 }
+                ("simulate-block-production", Some(arg_matches)) => {
+                    let marker_file = simulate_block_production_marker_path(&ledger_path);
+                    if !marker_file.exists() {
+                        eprintln!(
+                            "simulate-block-production is for development purposes only. \
+                             It's thus a pretty destructive operation on \
+                             the ledger ({ledger_path:?}). \
+                             Create an empty file at {marker_file:?} if this is intentional."
+                        );
+                        exit(1);
+                    }
+
+                    let mut process_options = parse_process_options(&ledger_path, arg_matches);
+
+                    let banking_trace_events = load_banking_trace_events_or_exit(&ledger_path);
+                    process_options.hash_overrides =
+                        Some(banking_trace_events.hash_overrides().clone());
+
+                    let slot = value_t!(arg_matches, "first_simulated_slot", Slot).unwrap();
+                    let simulator = BankingSimulator::new(banking_trace_events, slot);
+                    let Some(parent_slot) = simulator.parent_slot() else {
+                        eprintln!(
+                            "Couldn't determine parent_slot of first_simulated_slot: {slot} \
+                             due to missing banking_trace_event data."
+                        );
+                        exit(1);
+                    };
+                    process_options.halt_at_slot = Some(parent_slot);
```

Comment:
After writing up my other comment, questioned whether we should check if bank forks working bank slot is equal to `parent_slot`. This check will ensure we have a replayable path `>= halt_at_slot`, but not necessarily have `halt_at_slot` itself:
https://github.com/anza-xyz/agave/blob/c7e44c1389e01b576b1180387db20971eb1c8d3c/ledger-tool/src/ledger_utils.rs#L192-L193

Reply:
> we should check if bank forks working bank slot is equal to `parent_slot`

hehe, good point. this check is done here: https://github.com/anza-xyz/agave/pull/2733#discussion_r1752125145

I think panic is okay-ish. after all, this is dev subcommand..